lib/types/multipart.js

```diff
@@ -156,8 +157,10 @@ function Multipart(boy, cfg) {
           if (RE_NAME.test(parsed[i][0]))
             fieldname = decodeText(parsed[i][1], 'binary', 'utf8');
           else if (RE_FILENAME.test(parsed[i][0]))
-            filename = basename(decodeText(parsed[i][1], 'binary', 'utf8'));
+            filename = decodeText(parsed[i][1], 'binary', 'utf8');
         }
+        if (!preservePath)
```

Comment:
Either this line needs to include `&& filename` or these two lines need to be moved inside the `RE_FILENAME.test()` block after the `filename` is set.